src/IKeyValueAdapter.ts

```diff
@@ -5,12 +5,11 @@ import { ConfigurationSetting } from "@azure/app-configuration";
 export interface IKeyValueAdapter {
     /**
      * Determine whether the adapter applies to a configuration setting.
-     * Note: A setting is expected to be processed by at most one adapter.

```

Comment:
@avanigupta 
Here the use case is to store some correlated configs altogether into an object and JSON.stringify it. Maybe for some security reason, put it into a key vault, marking it as application/json. For JavaScript developers it looks a valid case. We should auto-parse it as we see the content type.

In dotnet, only one key value adapter would process a setting. But here to support the case, we should remove the constraints. The adapters should work like pipeline processors. We can rename it if you feel strange about the name "adapter".